assignments/arrays.js

```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+// console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
+const invSearch = (arr, invNum) => {
+   
+    for(i=0; i<arr.length; i++){
+        if(arr[i].id === invNum){
+            console.log(`Car ${arr[i].id} is a ${arr[i].car_year} ${arr[i].car_make} ${arr[i].car_model}`);
+        }
+    }
+};
+
+invSearch(inventory, 33);
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
-let lastCar = 0;
-console.log();
+let lastCar = inventory.length -1;
+console.log(`The last car has id ${inventory[lastCar].id} and is a ${inventory[lastCar].car_year} ${inventory[lastCar].car_make} ${inventory[lastCar].car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+
+// const sortAlpha = function(arr){
+//     for(let i=0; i<arr.length; i++){
+//         carModels.push(arr[i].car_make);
+//     };
+//     retrun carModels; 
+// }
+
+// const modelSort = (arr) => {
+//         let modelList = [];
+//         for(let i = 0; i<arr; i++){
+//             modelList.push(arr[i].car_model);
+//         }
+//         return modelList;
+//     };
+//  
+// carModels = modelSort(inventory);
+
+// let carModels =[];
+// for( let i=0; i<arr.length; i++){
+//     for (prop in arr[i]){
+//         carModels.push(arr[i].car_model)
+//     };
+// }
+
+
+
+// let carModels = inventory.sort((a,b)=>{
+//     let modelA = a.car_model.toLowerCase();
+//     let modelB = b.car_model.toLowerCase();
+//     if(modelA < modelB){
+//         return -1; 
+//     }
+// });
 let carModels = [];
-console.log();
+
+const sortModels = (arr) => {
+    for(let i = arr.length -1; i >= 0; i--){
+        for(let j = 1; j<=i; j++){
+            if(arr[j-1].car_model>arr[j].car_model){
+                let temp = arr[j-1];
+                arr[j-1] = arr[j];
+                arr[j] = temp;
+            }
+        }
+    }
+    return arr;
+}
```

Comment:
I feel you were getting in the weeds a little bit here and over-complicating things with all the functions leading up to this as well as your implementation. if a function ends up not working or being unused you can just delete it to make your project easier to read. you started to go in the right direction with `.sort()`, all you had to do here was iterate over the inventory and push all the car models into an empty array, apply `.sort()` with no parameters which automatically sorts it alphabetically, and console log the new array

Reply:
After 20min of playing with sort I went back to my playbook and used a bubble sort that I found online a while back.

Reply:
bubble sorts can be crazy and I'm glad to see you pushing to use something more complex like that, I just wanted you to be aware that there was a simpler solution :+1: